src/madsci_data_manager/tests/test_data_client.py

```diff
@@ -145,3 +154,398 @@ def test_local_only_dataclient(tmp_path: str) -> None:
+    assert uploaded_datapoint.object_name == f"direct_{file_path.name}", (
+        "Wrong object name"
+    )
+    assert uploaded_datapoint.custom_metadata.get("test_type") == "direct_submission", (
+        "Metadata not preserved"
+    )
+
+    # Download to a new location
+    download_path = tmp_path / f"downloaded_direct_{file_path.name}"
+    client.save_datapoint_value(uploaded_datapoint.datapoint_id, str(download_path))
+
+    # Verify download was successful
+    assert download_path.exists(), "Downloaded file doesn't exist"
+
+    # Check file contents and size
+    original_size = file_path.stat().st_size
+    downloaded_size = download_path.stat().st_size
+    assert downloaded_size == original_size, "File sizes don't match"
+
+    downloaded_content = download_path.read_text()
+    assert downloaded_content == file_content, "File contents don't match"
+
+    # Verify the URL format is correct
+    expected_url_prefix = f"http://{minio_server['host']}:{minio_server['port']}/madsci-test/direct_{file_path.name}"
+    assert uploaded_datapoint.url.startswith(expected_url_prefix), (
+        f"URL format incorrect. Expected to start with {expected_url_prefix}, got {uploaded_datapoint.url}"
+    )
+
+    # Verify etag exists
+    assert hasattr(uploaded_datapoint, "etag") and uploaded_datapoint.etag, (
+        "Missing etag"
+    )
+
+
+# Helper to clean up any leftover test containers
+def cleanup_test_containers():  # noqa
+    """Clean up any leftover MinIO test containers."""
+    try:
+        # List containers with our test prefix
+        result = subprocess.run(  # noqa
+            [  # noqa
+                "docker",
+                "ps",
+                "-a",
+                "--filter",
+                "name=minio_test_",
+                "--format",
+                "{{.Names}}",
+            ],
+            capture_output=True,
+            text=True,
+            timeout=10,
+            check=False,
+        )
+
+        if result.returncode == 0 and result.stdout.strip():
+            container_names = result.stdout.strip().split("\n")
+            for name in container_names:
+                if name.startswith("minio_test_"):
+                    print(f"Cleaning up leftover container: {name}")  # noqa
+                    subprocess.run(  # noqa
+                        ["docker", "stop", name],  # noqa
+                        capture_output=True,
+                        check=False,
+                    )  # noqa
+                    subprocess.run(  # noqa
+                        ["docker", "rm", name],  # noqa
+                        capture_output=True,
+                        check=False,
+                    )
+
+    except Exception as e:
+        print(f"Could not clean up test containers: {e}")  # noqa
+
+
+# Run cleanup before tests if needed
+@pytest.fixture(scope="session", autouse=True)
+def cleanup_before_tests():  # noqa
```

Comment:
Should probably also have a cleanup_after_tests so we're not leaving idle containers running around